source/opt/loop_descriptor.cpp

```diff
@@ -485,10 +485,24 @@ void Loop::ComputeLoopStructuredOrder(
 
   if (include_pre_header && GetPreHeaderBlock())
     ordered_loop_blocks->push_back(loop_preheader_);
-  cfg.ForEachBlockInReversePostOrder(
-      loop_header_, [ordered_loop_blocks, this](BasicBlock* bb) {
-        if (IsInsideLoop(bb)) ordered_loop_blocks->push_back(bb);
-      });
+
```

Comment:
Add a comment here describing why the different handling for shaders?  I'm not understanding how the shader case is dealing with the unreachable block.

Reply:
done.